src/hosts.rs

```diff
@@ -0,0 +1,323 @@
+use crate::{config::Config, messages};
+use core::fmt;
+use mysql::{prelude::Queryable, Conn, OptsBuilder};
+
+#[allow(dead_code)]
+/// Defines the possible status of a host
+#[derive(PartialEq, Clone, Copy)]
+pub enum HostStatus {
+    ///backend server is fully operational
+    Online,
+    //backend sever is temporarily taken out of use because of either too many connection errors in a time that was too short, or the replication lag exceeded the allowed threshold
+    Shunned,
+    //when a server is put into OFFLINE_SOFT mode, no new connections are created toward that server, while the existing connections are kept until they are returned to the connection pool or destructed. In other words, connections are kept in use until multiplexing is enabled again, for example when a transaction is completed. This makes it possible to gracefully detach a backend as long as multiplexing is efficient
```

Comment:
Seems like you might want to run `cargo fmt` to wrap these longer lines

Reply:
The longer lines do not format. Currently this project enforces cargo fmt already.
I can see we use a specific config to wrap comments on Readyset, however, those features are from nightly:

```
marcelo Work/readyset_proxysql_scheduler [query-search-method] $ cargo fmt --check
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `comment_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Module`, unstable features are only available in nightly channel.
Warning: can't set `group_imports = StdExternalCrate`, unstable features are only available in nightly channel.
```

Added the `///` to make it consistent.